build_tools/configure_stage.py

```diff
@@ -0,0 +1,227 @@
+        stage_name: Name of the build stage
+        amdgpu_families: Comma-separated GPU families
+        topology: BuildTopology instance
+        include_comments: Include comment lines explaining each flag
+
+    Returns:
+        List of CMake argument strings
+    """
+    args = []
+
+    if include_comments:
+        args.append(f"# CMake arguments for stage: {stage_name}")
+        args.append("")
+
+    # GPU families
+    if amdgpu_families:
+        args.append(f"-DTHEROCK_AMDGPU_FAMILIES={amdgpu_families}")
+
+    # Disable all features by default, then enable only what we need
+    if include_comments:
+        args.append("")
+        args.append("# Disable all features by default")
+    args.append("-DTHEROCK_ENABLE_ALL=OFF")
+
+    # Get features to enable for this stage
+    features = get_stage_features(topology, stage_name)
+
+    if include_comments:
+        args.append("")
+        args.append(f"# Enable features for stage '{stage_name}'")
+
+    for feature in sorted(features):
+        args.append(f"-DTHEROCK_ENABLE_{feature}=ON")
+
+    return args
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Generate CMake configuration for building a specific stage",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog=__doc__,
+    )
+    parser.add_argument(
+        "--stage",
+        type=str,
+        required=True,
+        help="Build stage name (e.g., foundation, compiler-runtime, math-libs)",
+    )
+    parser.add_argument(
+        "--amdgpu-families",
+        type=str,
+        default="",
+        help="Comma-separated GPU families (e.g., gfx94X-dcgpu,gfx110X-all)",
+    )
+    parser.add_argument(
+        "--output-cmake-args",
+        type=Path,
+        help="Output file for CMake arguments (one per line)",
+    )
+    parser.add_argument(
+        "--print",
+        action="store_true",
+        dest="print_args",
+        help="Print CMake arguments to stdout",
+    )
+    parser.add_argument(
+        "--comments",
+        action="store_true",
+        help="Include comments in output",
+    )
+    parser.add_argument(
+        "--oneline",
+        action="store_true",
+        help="Output all arguments on a single line (for shell expansion)",
+    )
+    parser.add_argument(
+        "--list-stages",
+        action="store_true",
+        help="List available build stages and exit",
+    )
+
+    args = parser.parse_args()
+
+    topology = get_topology()
+
+    # List stages mode
+    if args.list_stages:
+        log("Available build stages:")
+        for stage in topology.get_build_stages():
+            log(f"  {stage.name} ({stage.type}): {stage.description}")
+        return
+
+    # Validate stage
+    if args.stage not in topology.build_stages:
+        log(f"ERROR: Unknown stage '{args.stage}'")
+        log("Available stages:")
+        for stage in topology.get_build_stages():
+            log(f"  {stage.name}")
+        sys.exit(1)
```

Comment:
nit: This could use `parser.error()` instead of `sys.exit(1)`.

* Docs: https://docs.python.org/3/library/argparse.html#exiting-methods
* More tips: https://github.com/ROCm/TheRock/blob/main/docs/development/style_guide.md#use-argparse-for-cli-flags